apps/sharedtest/test/unit/crop_resize_rotate_test.js

```diff
@@ -4,13 +4,95 @@
 /* global parseJPEGMetadata */
 /* global getImageSize */
 /* global Downsample */
-/* global cropResizeRotate */
+/* global cropResizeRotate, JPEG */
 
+require('/shared/test/unit/mocks/mock_lazy_loader.js');
 require('/shared/js/blobview.js');
 require('/shared/js/media/jpeg_metadata_parser.js');
 require('/shared/js/media/image_size.js');
 require('/shared/js/media/downsample.js');
 require('/shared/js/media/crop_resize_rotate.js');
+require('/shared/js/media/jpeg-exif.js');
+
+function fetchBuffer(url) {
+  return new Promise(function(resolve, reject) {
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', url);
+    xhr.onload = function() {
+      if (xhr.status !== 200) {
+        reject(new Error('Failed with status: ' + xhr.status));
+      } else {
+        resolve(this.response);
+      }
+    };
+    xhr.onerror = xhr.ontimeout = function() {
+      reject(new Error('Failed' + xhr.status));
+    };
+    xhr.responseType = 'arraybuffer';
+    xhr.send();
+  });
+}
+
+function fetchBlob(url) {
+  return fetchBuffer(url).then(function(buffer) {
+    assert.ok(buffer);
+    return Promise.resolve(new Blob([buffer]));
+  }).catch(function(e) {
+    assert.ok(false, 'fetchBlob error' + e);
+  });
+}
+
+
+suite('Metadata copy test', function() {
+  suiteSetup(function(done) {
+
+    // We need the mock lazy loader for jpeg-exif.
+    window.LazyLoader = window.MockLazyLoader;
+    done();
+  });
+
+
+  [
+    { fileName: '/test-data/IMG_0001.jpg', dims: true, rotated: false },
+    { fileName: '/test-data/IMG_0010.jpg', dims: true, rotated: true }
+  ].
+    forEach(function (testFile) {
+      var testName = 'Testing metadata copy for ' + testFile.fileName;
+      test(testName, function(done) {
+
+        fetchBlob(testFile.fileName).then(function (blob) {
+          assert.ok(blob);
+          cropResizeRotate(
+            blob, null, { width: 640, height: 480 }, function (error, outBlob) {
+              assert.equal(error, null);
+              assert.ok(outBlob);
+
+              JPEG.readExifMetaData(outBlob, function (error, metaData) {
+                assert.ok(metaData);
+                // check metaData
+                assert.equal(metaData.Orientation, 1);
+                if (testFile.dims) {
+                  if (testFile.rotated) {
+                    assert.equal(metaData.PixelXDimension, 480);
+                    assert.equal(metaData.PixelYDimension, 640);
```

Comment:
I'm confused about this case. Aren't we asking for an output image that is always 640 wide by 480 high? In the case where cropResizeRotate alters the orientation metadata, shouldn't it also swap the PixelX/YDimensions metadata?